### ISSUE TITLE
Blank output with question prefix on inference with `test_git_inference_single_image`

### ISSUE THREAD

**Issue by micahscopes** [microsoft/GenerativeImage2Text#5]
I managed to get caption completion working alright (both in leaving the `prefix` text blank or passing in a partial caption for completion), but when I type a question of the form "this is a question?" (with a "?" at the end) I always get `output:` followed by nothing.

Similarly, I attempted using `[MASK]` tokens for a kind of fill-in-the-blank prompt and also get `output:` followed by nothing.

What am I missing?

**Comment by amsword** [comment]
if it is blank, it normally means it hits [EOS] token. 
- if what you tried out is GIT_BASE, it may be reasonable to have an empty output. The model is the pre-trained model, and the pre-training data has no VQA data. Only image-text pairs are available. The dataset is also described in the paper (Appendix). Thus, it is possible to output nothing. 
- If what you tried out is GIT_BASE_VQAv2, it is more likely to give a meaningful answer. For example of the following command line, the output on my side is boats.
```python
AZFUSE_TSV_USE_FUSE=1 python -m generativeimage2text.inference -p "{'type': 'test_git_inference_single_image', \
      'image_path': 'aux_data/images/1.jpg', \
      'model_name': 'GIT_BASE_VQAv2', \
      'prefix': 'what is it?', \
}"
```
- The model is pre-trained with language modeling tasks. There is no [MASK] token learned during pre-training and fine-tuning. Thus, if you give a prefix containing [MASK], the model itself cannot understand well what MASK here is. It is not masked language modeling task, which predicts the masked token.

**Comment by micahscopes** [comment]
wonderful, I'm pretty new to this stuff so thank you for your guidance

**Comment by micahscopes** [comment]
@amsword the output is still very terse, consistently a single word.  Is there a way to increase the temperature to get the models to rant a bit more?  I tried increasing the default temperature in the decoder and decreasing the length penalty to 0 but this doesn't seem to have any effect.

**Comment by amsword** [comment]
I assume what you are trying is the model with _VQAv2.
- the observation of terse answers is expected, because the dataset of VQAv2 contains only terse answers. The model behavior relies heavily on the training data and thus the prediction follows the style of the training data. 
- By default, the decoding is greedy-decoding without randomness. Changing the temperature should not give different results. To enable randomness, you can try to check if the parameter of do_sample is true
https://github.com/microsoft/GenerativeImage2Text/blob/bdc455b98a4bc9e20a3900be53d1018b6e8e02f3/generativeimage2text/layers/decoder.py#L226 
https://github.com/microsoft/GenerativeImage2Text/blob/bdc455b98a4bc9e20a3900be53d1018b6e8e02f3/generativeimage2text/layers/decoder.py#L1088

**Comment by micahscopes** [comment]
I managed to get this working and it did add some variation.  In case anyone else is curious to try, you also need to set the `top_k` parameter [here](https://github.com/microsoft/GenerativeImage2Text/blob/main/generativeimage2text/layers/decoder.py#L896-L897).

It seems that depending on how I set `top_k` and `temperature`, the results quickly become very unpredictable jibberish.

I'm guessing what I'm really after is to fine tune the models with some examples of longer, more meandering completions, which I've been avoiding.  @amsword do you think fine tuning would be the most straightforward way to influence the verbosity and length of the output?

**Comment by amsword** [comment]
> I managed to get this working and it did add some variation. In case anyone else is curious to try, you also need to set the `top_k` parameter [here](https://github.com/microsoft/GenerativeImage2Text/blob/main/generativeimage2text/layers/decoder.py#L896-L897).
> 
> It seems that depending on how I set `top_k` and `temperature`, the results quickly become very unpredictable jibberish.
> 
> I'm guessing what I'm really after is to fine tune the models with some examples of longer, more meandering completions, which I've been avoiding. @amsword do you think fine tuning would be the most straightforward way to influence the verbosity and length of the output?

Yes. I think so. The problem is the data collection which contains longer answers.
